### ISSUE TITLE
Allow users to save dot code from browser

### ISSUE THREAD

**Issue by DrQz** [mstefaniuk/graphviz.it#11]

**Comment by mstefaniuk** [comment]
What do you mean exactly? Using copy / paste is not enough?

**Comment by DrQz** [comment]
Like saving the image to a local file is nice button option, it seemed saving the text locally would be simple. It's a bit confusing from the current GUI as to what the Save options really are. Copy/paste is like saying, why not just do a local screen capture?

**Comment by mstefaniuk** [comment]
Ok, I can manage that.

**Comment by DrQz** [comment]
Cool. It might make things a bit clearer for a user if the "Update diagram" drop-down menu was labeled as "Save options" with the menu items: "Save PNG to disk" and "Save dot to disk" (or similar). I know I found "Update diagram" confusing since the graph canvas gets updated automatically with each edit in the text pane, so no other physical action is required.

**Comment by mstefaniuk** [comment]
Idea is to provide button with default action saving graphviz fiddle in the cloud and optional actions like saving as png, as dot source (as you requested) and maybe as SVG. Maybe naming should be improved "Update diagram fiddle" or "Save diagram to the cloud".

**Comment by DrQz** [comment]
When I first saw GV.it, I thought that it would be a good way to show the result to multiple people by sending a link to that page, as opposed to using the conventional GV app locally and sending the resulting image via email or Twitter, or whatever. Although GV.it doesn't work that way right now (does it?), it might be part of your plan with the cloud and so on. GV.it could also be used collaboratively, where 2 or more ppl update the diagram iteratively as part of a discussion or some other collaborative effort. That's why saving both the dot text and the image is important. In fact, **I could use such a feature right now, if it existed**. Point being, these things would clearly differentiate GV.it from vanilla GV. Just some random thoughts.

**Comment by mstefaniuk** [comment]
Thanks for pointing this out. It was one of my original idea for that project but lack of time slows me down. Please add another issue for that.

What do you think about naming button?

**Comment by DrQz** [comment]
No worries. I realize GV.it isn't providing your livelihood. :)

> What do you think about naming button?

As I mentioned previously, for the drop-down menu title (if that's what you're referring to), I believe  relabeling "Update diagram" to "Save options" would present the clearest meaning to any user. The word 'options' suggests to the user that they should click the _down arrow_ to see the option items (of which there's only one at the moment).

**Comment by mstefaniuk** [comment]
But it would make harder to save / update diagram fiddle to a cloud (two clicks instead of one). Currently it is main feature and saving PNG / DOT / SVG is secondary.

**Comment by DrQz** [comment]
Sorry, don't follow. As it is now, I have to issue 2 clicks: 1st to drop the menu, 2nd to save the png.  Not sure what you mean by "update diagram fiddle to a cloud is currently the main feature". In that case, I would expect to be able to leave GV.it, come back to it later and retrieve the most recent state. If that's the case, I don't know how to do it. Even if you know what you mean in terms of implementation, I'm  not sure a typical user gleans all that. Or maybe I'm missing something(?)

**Comment by mstefaniuk** [comment]
If you don't click on _down arrow_ then diagram would be saved to the cloud with permanent link. Then you can share it or return to it later.

**Comment by DrQz** [comment]
Right. I understand that now, but those semantics are not obvious at all, IMHO.
